### DISCUSSION TITLE
Train Models in Colab Best way to do

### DISCUSSION THREAD

**Discussion by sngraam** [SWivid/F5-TTS#466] [Ideas]
go to google colab 

connect type T4 (!important)

```
!apt-get update && apt-get install -y ffmpeg
!pip install gradio 
!pip uninstall -y torch torchvision torchaudio 
!pip install gradio datasets huggingface_hub
!pip install soundfile librosa
!pip install torch torchvision torchaudio --index-url  https://download.pytorch.org/whl/cu118
# 克隆并安装项目
!git clone https://github.com/thesandi99/sanddub.git
%cd sanddub
!pip install -e .
```
run your train finetune 

`!python finetune_gradio.py` make sure add in right dir.

check run `ls` insted `%cd src/f5_tts/train`

when you run transcript and others and now your turn to run `start traning`

stop gradio.py and add this code 

`!accelerate launch \
  --num_processes 1 \
  --num_machines 1 \
  --mixed_precision fp16 \
  --dynamo_backend no \
  /content/sanddub/src/f5_tts/train/finetune_cli.py \
  --exp_name F5TTS_Base \
  --learning_rate 1e-05 \
  --batch_size_per_gpu 400 \
  --batch_size_type frame \
  --max_samples 64 \
  --grad_accumulation_steps 1 \
  --max_grad_norm 1 \
  --epochs 11 \
  --num_warmup_updates 200 \
  --save_per_updates 400 \
  --last_per_steps 800 \
  --dataset_name my_speak \
  --finetune True \
  --tokenizer char \
  --log_samples True \
  --logger wandb
`

this is best setting i will check 100s of time ;} 

if your model done mesg will appear Saved last checkpoint at step 1600 make sure is right or not run further now run agin gradio.py and test your model

share... ;)

**Comment by mikhail2013ru** [comment]
How many epochs does it take to train 25 hours of dataset? Is the model's training supported in Russian? Is it possible to teach on kaggle too? As in Notepad to write the save the last epoch on Google drive or Yandex disk upon completion of 12 hours of active session? How do I resume training from the last era? Developers can add to the web interface, the ability to save the era of cloud Google or Yandex and resume training the model after the end of active session on kaggle or Colaba? Please add in the web interface download from Google drive or Yandex drive of dataset in a zip archive, and extract.

**Reply by MrxDoc** [reply]
Hi can u tell me, how can I use this in Google colab, bcz I have no GPU, so help me out guys, I'm not even a coding person so help me to achieve this in Google colab or else can u suggest me some youtube references I can watch it and set this up and i want to generate my sample voices to create multi emotions in single audio ?? help me out plz..
